commit 7560db9049bde7843cf1a1e840072aecf13d19b1
Author: Umaro <umaro@narshe.net>
Date:   Thu Jun 23 19:47:39 2011 +0200

    msm: vidc: Support dynamic change of interlaced format.
    
    The interlaced format can change dynamically with in a
    sequence/clip. This change ensures that the interlaced
    format is updated and passed to the client appropriately.
    
    Signed-off-by: Deepika Pepakayala <deepikap@codeaurora.org>
    (cherry picked from commit 7c395a949da6a08bb3a86a5faf760f26cbbd91db)
    
    Change-Id: I1bf81748245e34d33bc2a2f8746551f397ea0ed1
    Signed-off-by: Shruthi Krishna <skrish@codeaurora.org>

diff --git a/drivers/video/msm/vidc/common/dec/vdec.c b/drivers/video/msm/vidc/common/dec/vdec.c
index 07276de..6113284 100644
--- a/drivers/video/msm/vidc/common/dec/vdec.c
+++ b/drivers/video/msm/vidc/common/dec/vdec.c
@@ -259,6 +259,15 @@ static void vid_dec_output_frame_done(struct video_client_ctx *client_ctx,
 			vcd_frame_data->dec_op_prop.disp_frm.right;
 		vdec_msg->vdec_msg_info.msgdata.output_frame.framesize.top =
 			vcd_frame_data->dec_op_prop.disp_frm.top;
+		if (vcd_frame_data->interlaced) {
+                        vdec_msg->vdec_msg_info.msgdata.
+                                output_frame.interlaced_format =
+                                VDEC_InterlaceInterleaveFrameTopFieldFirst;
+                } else {
+                        vdec_msg->vdec_msg_info.msgdata.
+                                output_frame.interlaced_format =
+                                VDEC_InterlaceFrameProgressive;
+                }
 		/* Decoded picture type */
 		switch (vcd_frame_data->frame) {
 		case VCD_FRAME_I:
diff --git a/include/linux/msm_vidc_dec.h b/include/linux/msm_vidc_dec.h
index b12d51a..bcceac0 100644
--- a/include/linux/msm_vidc_dec.h
+++ b/include/linux/msm_vidc_dec.h
@@ -532,6 +532,7 @@ struct vdec_output_frameinfo {
 	void *client_data;
 	void *input_frame_clientdata;
 	struct vdec_framesize framesize;
+	enum vdec_interlaced_format interlaced_format;
 };
 
 union vdec_msgdata {
